File: trunk/emsi/asp-net-app/WebEMSOF/component/biz/Class_biz_user.pas
unit Class_bc_user;

interface

uses
  appcommon,
  Class_dalc_user,
  system.web;

const
  ID = '$Id$';

type
  TClass_bc_user = class
  private
    { Private Declarations }
  public
    constructor Create;
    function Kind: string;
    function IdNum: string;
    function Roles: appcommon.string_array;
  end;

implementation

const
  KIND_THAT_HAS_ROLES = 'regional_staffer';

constructor TClass_bc_user.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

function TClass_bc_user.Kind: string;
var
  name: string;
begin
  name := httpcontext.current.user.identity.name;
  Kind := name.Substring(0,name.LastIndexOf('_'));
end;

function TClass_bc_user.IdNum: string;
var
  name: string;
begin
  name := httpcontext.current.user.identity.name;
  IdNum := name.Substring(name.LastIndexOf('_') + 1);
end;

function TClass_bc_user.Roles: appcommon.string_array;
var
  name: string;
begin
  name := httpcontext.current.user.identity.name;
  if Kind = KIND_THAT_HAS_ROLES then begin
    Roles := Class_dalc_user.TClass_dalc_user.Create.RolesOf('regional_staffer_',IdNum);
  end;
end;

end.

File: asp-net-app/WebEMSOF/component/biz/Class_biz_user.cs
using Class_db_user;
using kix;
using System;
using System.Web;

namespace Class_biz_user
{
    public class TClass_biz_user
    {
        //Constructor  Create()
        public TClass_biz_user() : base()
        {
            // TODO: Add any constructor code here

        }
        public string Kind()
        {
            string result;
            string name;
            name = HttpContext.Current.User.Identity.Name;
            result = name.Substring(0, name.LastIndexOf("_"));
            return result;
        }

        public string IdNum()
        {
            string result;
            string name;
            name = HttpContext.Current.User.Identity.Name;
            result = name.Substring(name.LastIndexOf("_") + 1);
            return result;
        }

        public string[] Roles()
        {
            string[] result = {k.EMPTY};
            string name;
            name = HttpContext.Current.User.Identity.Name;
            if (Kind() == Units.Class_biz_user.KIND_THAT_HAS_ROLES)
            {
                result = new TClass_db_user().RolesOf("regional_staffer_", IdNum());
            }
            return result;
        }

    } // end TClass_biz_user

}

namespace Class_biz_user.Units
{
    public class Class_biz_user
    {
        public const string ID = "$Id: Class_biz_user.pas 2343 2008-05-28 21:19:37Z kevinanlipscomb $";
        public const string KIND_THAT_HAS_ROLES = "regional_staffer";
    } // end Class_biz_user

}

